### ISSUE TITLE
1.10 Bedrock Docs - Add a guide on how to style the style guide to your specific project

### ISSUE THREAD

**Issue by Wolfr** [usebedrock/bedrock#201]

**Comment by Wolfr** [comment]
Basically you need to set up variables according to your project's CSS.

Here is an example for Bootstrap:

````block pageVariables
    //- Use this block to append page variables in templates
    - var projectTitle = "Bedrock + Bootstrap 4"

    - var htmlClass = ""
    - var bodyClass = ""

    //- Styleguide layout system
    //- Use your project's layout system for the style guide
    
    //- This is usually a grid container like .container in Bootstrap, .o-grid in an ITCSS system etc.
    - var styleguideGridClass = "container"
    
    //- This is an optional child of the grid container like .row in Bootstrap
    - var styleguideGridInnerClass = "row"

    //- This is a generic class of a column col-md in Bootstrap 3 (not needed in Bootstrap 4)
    - var styleguideGridCellClass = "col-md"

    //-
        This is the sizing mechanism for the left or right columns in styleguide, e.g. col-md-3 for the left side and
        col-md-9 for the right side if your site uses 12 columns. Adjust accordingly depending on your grid system.
    - var styleguideLayoutTwoColumnsLeftClass = "col-md-3"
    - var styleguideLayoutTwoColumnsRightClass = "col-md-9"

    //- Styleguide classes - Buttons
    //- Style buttons like your project's buttons
    - var styleguideButtonClass = "btn btn-link btn-sm"

    //-
        Styleguide classes - Headings
        Style headings in styleguide like your project's headings
    - var styleguideBrandClass = "h4 mb-4"
    - var styleguideComponentGroupHeaderClass = "h3"
    - var styleguideSampleHeadingClass = "h4"
    - var styleguideComponentListingCategoryClass = "h4"

    - var styleguideNavListTitleClass = "h6"
    - var styleguideNavListClass = "list-group mb-4"
    - var styleguideSampleClass = "mb-4 mt-4"

    - var styleguideNavListItemLinkClass = "list-group-item list-group-item-action"
    - var styleguideNavListItemLinkActiveClass = "active"
    - var styleguideNavListItemActiveClass = ""
    - var styleguideNavListItemClass = ""

    - var styleguideNavigationHolderClass = "mr-md-4"
````

**Comment by Wolfr** [comment]
This should be tested and improved upon. Then we can add it to the docs on the website.

**Comment by Wolfr** [comment]
There could be an article

Making your own Bedrock base

* Decide your config
* Do some basic documentation of your components
* Set up styles

**Comment by Wolfr** [comment]
Here's a text that we need to add to the website,

text needs to be combined with the existing text at https://bedrockapp.org/documentation/styling-the-style-guide/ 


**Styling the styleguide** 

Here's the problem: bedrock supports any kind of CSS, but as framework author we have no idea whether

* You are using a CSS reset or normalize
* Whether you have spacing on your headings by default or that they need a class
* Whether your button styling is appropriate for style guide purposes

So, when creating a custom Bedrock install we asks you to customize Bedrock according to your needs.

We will create a small tutorial video that shows how to set up a custom Bedrock base at some point.

Basically it involves entering the right CSS classes into variables, and then maybe doing some additional styling based on your CSS by adding additional styles to styleguide.css.

This way, you keep your base CSS clean and devoid of any styles that are necessary for the “correct” display in Bedrock.

**Comment by Wolfr** [comment]
Revised text

> 
> **Styling the styleguide** 
> 
> Here's the problem: Bedrock supports any kind of CSS, but as a framework author we have no idea whether
> 
> * You are using a CSS reset or normalize
> * Whether you have spacing on your headings by default or that they need a class to get styling
> * Whether your button styling is appropriate for style guide purposes
> 
> When creating a Bedrock prototype you can add some classes to easily style the style guide depending on your project's needs.
> Basically it involves entering the right CSS classes into variables, and then maybe doing some additional styling based on your CSS by adding additional styles to styleguide.css (depending on how your code works).
> 
> This way, you keep your base CSS clean and devoid of any styles that are necessary for the “correct” display in Bedrock.
> 
>
